File: GCD.c
#include <stdio.h>
int gcd(int a,int b)
{
	if (a%b==0){
		return(b);
	}
	else{
		return(gcd(b,a%b));
	}
}

int main(){
	int x=0;
	int y=0;
	printf("Enter a positive integer: ");
	scanf("%d", &x);
	x = scanf("%d", &x);
	

	printf("Enter another positive integer: ");
	scanf("%d", &y);
	y = scanf("%d", &y);
	printf("The GCD of %d and %d is: %d ",x,y,gcd(x,y));
	return 0;
}

File: GCD.java
import java.util.Scanner;
public class GCD {
    public static void main(String [] args){
	Scanner sc = new Scanner(System.in);

	System.out.println("Enter a positive integer: ");
	while( !sc.hasNextDouble() ){
		sc.next();
		System.out.println("Please enter a positive integer: ");
	}

	int a = sc.nextInt();
	System.out.println("Enter another postive integer: ");
	while( !sc.hasNextInt() ){
	    sc.next();
	    System.out.println("Please enter a positive integer: ");
	}

        int b = sc.nextInt();
         
        System.out.println("The GCD of " + a + " and " + b + " is " + getGCD(a,b) + ".");
     }
     
     public static int getGCD(int a, int b){
	int GCD = 1;

        if(a>b){
            for(int i = b; i >=1; i--){
                if(a%i==0 && b%i==0){
                return i;
		}
            }
	}
	else{
	  for(int j = a; j >=1; j--){
		if(a%j==0 && b%j==0){
		return j;
	        }
             }
	}
	return GCD;
     }
}
